pubnub-gson/src/integrationTest/java/com/pubnub/api/integration/util/BaseIntegrationTest.java

```diff
@@ -0,0 +1,388 @@
+
+        pubnub.addListener(new SubscribeCallback() {
+            @Override
+            public void file(@NotNull PubNub pubnub, @NotNull PNFileEventResult pnFileEventResult) {
+
+            }
+
+            @Override
+            public void status(@NotNull PubNub pubnub, @NotNull PNStatus status) {
+                if (status.getCategory() == PNStatusCategory.PNConnectedCategory) {
+                    if (status.getAffectedChannels().containsAll(channels)) {
+                        subscribeSuccess.set(true);
+                    }
+                } else if (status.getCategory() == PNStatusCategory.PNSubscriptionChanged) {
+                    if (status.getAffectedChannels().containsAll(channels)) {
+                        subscribeSuccess.set(true);
+                    }
+                }
+            }
+
+            @Override
+            public void message(@NotNull PubNub pubnub, @NotNull PNMessageResult message) {
+
+            }
+
+            @Override
+            public void presence(@NotNull PubNub pubnub, @NotNull PNPresenceEventResult presence) {
+
+            }
+
+            @Override
+            public void signal(@NotNull PubNub pubNub, @NotNull PNSignalResult pnSignalResult) {
+
+            }
+
+            @Override
+            public void uuid(@NotNull final PubNub pubnub, @NotNull final PNUUIDMetadataResult pnUUIDMetadataResult) {
+
+            }
+
+            @Override
+            public void channel(@NotNull final PubNub pubnub, @NotNull final PNChannelMetadataResult pnChannelMetadataResult) {
+
+            }
+
+            @Override
+            public void membership(@NotNull final PubNub pubnub, @NotNull final PNMembershipResult pnMembershipResult) {
+
+            }
+
+            @Override
+            public void messageAction(@NotNull PubNub pubnub, @NotNull PNMessageActionResult pnActionResult) {
+
+            }
+
+        });
+
+        pubnub.subscribe()
+                .channels(channels)
+                .withPresence()
+                .execute();
+
+        Awaitility.await().atMost(Durations.TEN_SECONDS).untilTrue(subscribeSuccess);
+    }
+
+    protected void subscribeToChannelGroup(@NotNull PubNub pubnub, @NotNull String group) {
+        pubnub.subscribe()
+                .channelGroups(Collections.singletonList(group))
+                .withPresence()
+                .execute();
+        pause(1);
+    }
+
+    protected void unsubscribeFromChannel(PubNub pubNub, String channel) {
+        pubNub.unsubscribe()
+                .channels(Collections.singletonList(channel))
+                .execute();
+        pause(1);
+    }
+
+    protected void unsubscribeFromAllChannels(PubNub pubNub) {
+        pubNub.unsubscribeAll();
+        pause(1);
+    }
+
+    protected Map<String, String> generateMessage(PubNub pubNub, String message) {
+        final Map<String, String> map = new HashMap<>();
+        map.put("publisher", pubNub.configuration.getUserId().getValue());
```

Comment:
I am getting:
"Cannot resolve symbol 'configuration'"
Does it work on your side?